core/modules/ccontrol/UserQuerySelect.cc

```diff
@@ -169,7 +169,12 @@ void UserQuerySelect::kill() {
     if (!_killed) {
         _killed = true;
         try {
-            _executive->squash();
+            // make a copy of executive pointer to keep it alive and avoid race
+            // with pointer being reset in discard() method
+            std::shared_ptr<qdisp::Executive> exec = _executive;
```

Comment:
Could use auto here.

Reply:
I wanted to make it very explicit that a copy of a shared pointer is needed here :)